src/transformation/visitors/function.ts

```diff
@@ -24,19 +24,26 @@ import { transformBindingPattern } from "./variable-declaration";
 function transformParameterDefaultValueDeclaration(
     context: TransformationContext,
     parameterName: lua.Identifier,
-    value?: ts.Expression,
+    value: ts.Expression,
     tsOriginal?: ts.Node
-): lua.Statement {
-    const parameterValue = value ? context.transformExpression(value) : undefined;
-    const assignment = lua.createAssignmentStatement(parameterName, parameterValue);
+): lua.Statement | undefined {
+    // const parameterValue = context.transformExpression(value);
+    const { precedingStatements, result: parameterValue } = transformInPrecedingStatementScope(context, () =>
+        context.transformExpression(value)
+    );
+    const statements = [...precedingStatements];
+    if (!lua.isNilLiteral(parameterValue)) {
```

Comment:
Couldn't we do that at the top of the function? Check ts.isUndefinedLiteral || ts.isNullLiteral? That way we can avoid a whole bunch of transformations.

Reply:
Doing this makes this the most general; it applies to anything that results in a nil literal (casts/parenthesis/etc, language extensions, plugin output) and shouldn't have any performance impact.